src/pint/utils.py

```diff
@@ -1258,13 +1288,76 @@ def companion_mass(pb, x, inc=60.0 * u.deg, mpsr=1.4 * u.solMass):
     -------
     mass : Quantity in u.solMass
     """
-    massfunct = mass_funct(pb, x)
+    if not (isinstance(inc, angles.Angle) or isinstance(inc, u.quantity.Quantity)):
+        raise ValueError(f"The inclination should be an Angle but is {inc}.")
+    if not isinstance(x, u.quantity.Quantity):
+        raise ValueError(
+            f"The projected semi-major axis x should be a Quantity but is {x}."
+        )
+    if not isinstance(pb, u.quantity.Quantity):
+        raise ValueError(f"The binary period pb should be a Quantity but is {pb}.")
+    if not isinstance(mpsr, u.quantity.Quantity):
+        raise ValueError(f"The pulsar mass should be a Quantity but is {mpsr}.")
+
+    """
```

Comment:
Is it worth putting this in a Note section of the docstring?

Reply:
OK, just pushed up changes with everything in the docstrings